### PR TITLE
Environmentally friendly tests

### PR DESCRIPTION
**Pull request by felixmulder**

### CODE REVIEW

File: compiler/test/dotc/tests.scala [hunk 1]
```diff
@@ -338,10 +349,16 @@ class tests extends CompilerTest {
   @Test def tasty_tests = compileDir(testsDir, "tasty", testPickling)
 
   @Test def tasty_bootstrap = {
+    val f = new JFile(getClass.getProtectionDomain.getCodeSource.getLocation.getPath)
+    println(f)
+    println(System.getProperty("java.class.path"))
+
```
Comment (felixmulder) [inline compiler/test/dotc/tests.scala]: Note to self: get rid of this addition

File: compiler/test/dotc/tests.scala [hunk 2]
```diff
@@ -32,18 +33,28 @@ class tests extends CompilerTest {
...
+            | - DOTTY_LIBRARY
+            | - DOTTY_COMPILER
+            | - DOTTY_INTERFACES
+            | - DOTTY_EXTRAS
+            |
+            |Where these all contain locations, except extras which is a comma
+            |separated list of jars.
+            |
+            |When compiling with eclipse, you need the sbt-interfaces jar, but
+            |it in extras."""
```
Comment (felixmulder) [inline compiler/test/dotc/tests.scala]: *put

**Comment by felixmulder** [comment]
@smarter - would you mind reviewing?

File: compiler/test/dotc/tests.scala [hunk 3]
```diff
@@ -32,18 +33,28 @@ class tests extends CompilerTest {
...
+            |testing.
+            |
+            |If running without sbt, test paths need to be setup environment variables:
+            |
+            | - DOTTY_LIBRARY
+            | - DOTTY_COMPILER
+            | - DOTTY_INTERFACES
+            | - DOTTY_EXTRAS
+            |
+            |Where these all contain locations, except extras which is a comma
```
Comment (smarter) [inline compiler/test/dotc/tests.scala]: colon separated now

**Pull request merged**